SSA/Core/Framework.lean

```diff
@@ -522,24 +522,23 @@ end Lemmas
 ## `denote`
 Denote expressions, programs, and sequences of lets
 -/
-variable [TyDenote d.Ty] [DialectDenote d] [DecidableEq d.Ty] [Monad d.m] [LawfulMonad d.m]
```

Comment:
the change here seems very strong, it seems like we're using the typeclasses `TyDenote`, `Monad` and `DialectDenote` essentially everywhere  - shouldn't these stay as variables here? We can use sections to delimit these if there are some theorems that don't depend on them, but at least from a first glance I couldn't find examples, and even if there was a few, I think we'd still prefer manually excluding those cases since these typeclasses are so ubiquitous.